frontend/utils/time.js

```diff
@@ -113,14 +113,20 @@ export function firstDayOfMonth (date: Date): Date {
   return new Date(date.getFullYear(), date.getMonth(), 1)
 }
 
-// TODO: Provide locale fallback in case navigator does not exist (e.g. server, Mocha, etc...)
-const locale = (typeof navigator === 'undefined' && 'en-US') || (navigator.languages ? navigator.languages[0] : navigator.language)
-
 export function humanDate (
   date: number | Date | string,
-  opts?: Intl$DateTimeFormatOptions = { month: 'short', day: 'numeric' }
+  options?: Intl$DateTimeFormatOptions = { month: 'short', day: 'numeric' }
 ): string {
-  return new Date(date).toLocaleDateString(locale, opts)
+  const locale = typeof navigator === 'undefined'
+    // Fallback for Mocha tests.
+    ? 'en-US'
+    // Flow considers `navigator.languages` to be of type `$ReadOnlyArray<string>`,
+    // which is not compatible with the `string[]` expected by `.toLocaleDateString()`.
+    // Casting to `string[]` through `any` as a workaround.
+    : ((navigator.languages: any): string[]) ?? navigator.language
+  // NOTE: `.toLocaleDateString()` automatically takes local timezone differences into account.
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString
```

Comment:
OK, essentially the same thing except we're allowing the full array to be passed in. That seems fine. However, why did you move this into this function instead of leaving it outside of it? Now this code will have to be run each time `humanDate` is called, and that seems unnecessary.

Reply:
Mainly because other functions in this file don't need access to that variable.
I agree recomputing it is unnecessay most of the time, unless the user change their browser locale.